Containers/sg_StaticMap.hpp

```diff
@@ -376,12 +378,25 @@ juce::Array<KeyType> StaticMap<KeyType, ValueType, Capacity>::getKeys() const no
     JUCE_ASSERT_MESSAGE_THREAD;
     juce::Array<KeyType> result{};
     result.ensureStorageAllocated(narrow<int>(size()));
-    for (auto const node : *this) {
+    for (auto const & node : *this) {
         result.add(node.key);
     }
     return result;
 }
 
+//==============================================================================
+template<typename KeyType, typename ValueType, size_t Capacity>
+std::vector<KeyType> StaticMap<KeyType, ValueType, Capacity>::getKeyVector() const noexcept
+{
```

Comment:
IDK if we have access to ranges in SpatGRIS - if so, I think maybe this function could become (assuming there's a ValueType defined): 

    JUCE_ASSERT_MESSAGE_THREAD;
    return *this | std::views::transform(&ValueType::key) | std::ranges::to<std::vector>();
    
example: https://gcc.godbolt.org/z/c8qsenbKc

Reply:
neat refactor otherwise :D

Reply:
ah well, it'll be for spatgris 3!